models/review.go

```diff
@@ -389,6 +413,31 @@ func GetReviewersByIssueID(issueID int64) (reviews []*Review, err error) {
 	return reviews, nil
 }
 
+// GetReviewerByIssueIDAndUserID get the latest review of reviewer for a pull request
+func GetReviewerByIssueIDAndUserID(issueID, UserID int64) (review *Review, err error) {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return nil, err
+	}
+
+	var reviewsUnfiltered []*Review
+
+	if err := sess.SQL("SELECT * FROM review WHERE id IN (SELECT max(id) as id FROM review WHERE issue_id = ? AND reviewer_id = ? AND type in (?, ?, ?) GROUP BY issue_id, reviewer_id) ORDER BY review.updated_unix ASC",
```

Comment:
Why is the `GROUP BY` required? `max(id)` should be enough since you're not including any scalar column.

Also, you could use `x. ... .Get(&review)` instead of `x. ... .Find(...)` to get a single row. That will also check that there's only one result for you.